src/main/java/org/apache/cassandra/sidecar/routes/TokenRangeReplicaMapHandler.java

```diff
@@ -0,0 +1,114 @@
+import org.apache.cassandra.sidecar.concurrent.ExecutorPools;
+import org.apache.cassandra.sidecar.utils.InstanceMetadataFetcher;
+
+import static org.apache.cassandra.sidecar.utils.HttpExceptions.cassandraServiceUnavailable;
+
+/**
+ * Handler which provides token range to read and write replica mapping
+ *
+ * <p>This handler provides token range replicas along with the state of the replicas. For the purpose
+ * of identifying the state of a newly joining node to replace a dead node from a newly joining node,
+ * a new state 'Replacing' has been added.
+ * It is represented by
+ * {@code org.apache.cassandra.sidecar.adapters.base.TokenRangeReplicaProvider.StateWithReplacement}
+ */
+@Singleton
+public class TokenRangeReplicaMapHandler extends AbstractHandler<TokenRangeReplicasRequest>
+{
+
+    @Inject
+    public TokenRangeReplicaMapHandler(InstanceMetadataFetcher metadataFetcher,
+                                       CassandraInputValidator validator,
+                                       ExecutorPools executorPools)
+    {
+        super(metadataFetcher, executorPools, validator);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handleInternal(RoutingContext context,
+                               HttpServerRequest httpRequest,
+                               String host,
+                               SocketAddress remoteAddress,
+                               TokenRangeReplicasRequest request)
+    {
+        CassandraAdapterDelegate delegate = metadataFetcher.delegate(host);
+
+        StorageOperations storageOperations = delegate.storageOperations();
+        Metadata metadata = delegate.metadata();
+        if (storageOperations == null || metadata == null)
+        {
+            context.fail(cassandraServiceUnavailable());
+            return;
+        }
+
+        executorPools.service().executeBlocking(promise -> {
+            try
+            {
+                context.json(storageOperations.tokenRangeReplicas(request.keyspace(), metadata.getPartitioner()));
+            }
+            catch (UnknownHostException e)
+            {
+                processFailure(e, context, host, remoteAddress, request);
+            }
+        }).onFailure(cause -> processFailure(cause, context, host, remoteAddress, request));
+    }
+
+    @Override
+    protected TokenRangeReplicasRequest extractParamsOrThrow(RoutingContext context)
+    {
+        return new TokenRangeReplicasRequest(keyspace(context, true));
+    }
+
+    @Override
+    protected void processFailure(Throwable cause, RoutingContext context, String host, SocketAddress remoteAddress,
+                                  TokenRangeReplicasRequest request)
+    {
+        if (cause instanceof AssertionError &&
+            StringUtils.contains(cause.getMessage(), "Unknown keyspace"))
+        {
+            context.fail(new HttpException(HttpResponseStatus.NOT_FOUND.code(), cause.getMessage()));
```

Comment:
nit, since we have these helper methods:
```suggestion
            context.fail(HttpExceptions.wrapHttpException(HttpResponseStatus.NOT_FOUND, cause.getMessage()));
```